main.go

```diff
@@ -0,0 +1,382 @@
+	}
+
+	thanosConfigMapName := os.Getenv("THANOS_CONFIGMAP_NAME")
+	if len(thanosConfigMapName) == 0 {
+		return errors.Errorf("THANOS_CONFIGMAP_NAME environment variable is not set.")
+	}
+
+	mattermostAlertsHook := os.Getenv("MATTERMOST_ALERTS_HOOK")
+	if len(mattermostAlertsHook) == 0 {
+		return errors.Errorf("MATTERMOST_ALERTS_HOOK environment variable is not set.")
+	}
+	return nil
+}
+
+// thanosStoreDiscovery is used to keep Thanos up to date with deployed Thanos Query targets.
+func thanosStoreDiscovery() error {
+	privateHostedZoneID := os.Getenv("PRIVATE_HOSTED_ZONE_ID")
+	thanosNamespace := os.Getenv("THANOS_NAMESPACE")
+	thanosDeploymentName := os.Getenv("THANOS_DEPLOYMENT_NAME")
+	thanosConfigMapName := os.Getenv("THANOS_CONFIGMAP_NAME")
+
+	log.Infof("Getting Route53 records for hostedzone %s", privateHostedZoneID)
+	records, err := listAllRecordSets(privateHostedZoneID)
+	if err != nil {
+		return errors.Wrap(err, "Unable to get the existing Thanos Route53 records")
+	}
+
+	thanosRecords := getThanosTargets(records)
+	if len(thanosRecords) < 1 {
+		log.Info("No Thanos records to register, canceling run")
+		return nil
+	}
+	route53Targets := addPortToTargets(thanosRecords)
+
+	// This part can be used for local testing
+	// kubeconfig := filepath.Join(
+	// 	os.Getenv("HOME"), ".kube", "config",
+	// )
+
+	// config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	// if err != nil {
+	// 	return err
+	// }
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return errors.Wrap(err, "Unable to set k8s access config")
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "Unable to create k8s clientset")
+	}
+
+	configMapTargets, err := getConfigMapTargets(thanosConfigMapName, thanosNamespace, clientset)
+	if err != nil && !errors.IsNotFound(err) {
+		return errors.Wrap(err, "Unable to get existing configmap targets")
+	}
+
+	configMapUpToDate := checkIfConfigMapUpToDate(configMapTargets, route53Targets)
+
+	if len(configMapTargets) < 1 || !configMapUpToDate {
+		log.Info("The Thanos configmap targets are not up to date with existing Route53 Thanos targets")
+		err = createOrUpdateTargetConfigMap(route53Targets, thanosConfigMapName, thanosNamespace, clientset)
+		if err != nil {
+			return errors.Wrap(err, "Unable to create or update the Thanos Query configmap")
+		}
+
+		err = rotateQueryPods(thanosNamespace, thanosDeploymentName, clientset)
+		if err != nil {
+			return errors.Wrap(err, "Unable to rotate Thanos Query pods")
+		}
+
+		log.Info("Successfully rotated all Thanos Query pods")
+	} else {
+		log.Info("The Thanos configmap targets are up to date with existing Route53 Thanos targets")
+	}
+	return nil
+}
+
+// checkIfConfigMapUpToDate is used to check if the current configmap targets are up to date with the deploye Route53 targets.
+func checkIfConfigMapUpToDate(configMapTargets, route53Targets []string) bool {
```

Comment:
The func name could be `isConfigMapUpToDate` in order for the boolean response to make sense 
```suggestion
func isConfigMapUpToDate(configMapTargets, route53Targets []string) bool {
```

Reply:
I like that more, thanks :)